### DISCUSSION TITLE
Why is document upload size is 50MB max?

### DISCUSSION THREAD

**Discussion by silincone** [reo7sp/tgbot-cpp#296] [Q&A] [answered]
As I have mentioned, why is the upload size so less when telegram allows 2GB + 2GB* ? 
Or, am I just wrong with my conclusion? If I am then, can you please let me know the way to do it?
And, I have very little knowledge about programming, I couldn't understand the sample echo program, so that's where I stand, I hope you understand me and I hope that you could help me out.
Thank you for your time.

**Comment by reo7sp** [comment, marked as answer]
The 50 MB limit is imposed by the hosted Telegram Bot API, not by tgbot-cpp. There are three distinct cases:

1. Pass an existing Telegram `file_id`: Telegram reuses the stored file, so it is not uploaded again.
2. Upload a new file to `api.telegram.org`: general files are limited to 50 MB.
3. Run the official local `telegram-bot-api` server: uploads are supported up to 2000 MB and downloads have no size limit.

For the local server, point the third `Bot` constructor argument to it:

```cpp
TgBot::CurlHttpClient httpClient;
TgBot::Bot bot(token, httpClient, "http://localhost:8081");
```

`localhost:8081` is only an example; the official server must be installed, configured and listening on that address.
